yb-voyager/cmd/analyzeSchema.go

```diff
@@ -394,12 +396,80 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
+		here again similar to ALTER table Storage parameters options is the high level field in for WITH options.
+	*/
+	if len(createIndexNode.IndexStmt.GetOptions()) > 0 {
+		reportCase(fpath, STORAGE_PARAMETERS_DDL_STMT_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/23467",
+			"Remove the storage parameters from the DDL", "INDEX", indexName, sqlStmtInfo.stmt)
+	}
+}
+
+func reportAlterTableVariants(alterTableNode *pg_query.Node_AlterTableStmt, sqlStmtInfo sqlInfo, fpath string, objType string) {
+	schemaName := alterTableNode.AlterTableStmt.Relation.Schemaname
+	tableName := alterTableNode.AlterTableStmt.Relation.Relname
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	// this will the list of items in the SET (attribute=value, ..)
+	/*
+		e.g. alter table test_1 alter column col1 set (attribute_option=value);
+		cmds:{alter_table_cmd:{subtype:AT_SetOptions name:"col1" def:{list:{items:{def_elem:{defname:"attribute_option"
+		arg:{type_name:{names:{string:{sval:"value"}} typemod:-1 location:263}} defaction:DEFELEM_UNSPEC location:246}}}}...
+		for set attribute issue we will the type of alter setting the options and in the 'def' definition field which has the
+		information of the type, we will check if there is any list which will only present in case there is syntax like <SubTYPE> (...)
+	*/
+	setParameters := alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetDef().GetList()
+	if alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetSubtype() == pg_query.AlterTableType_AT_SetOptions &&
+		len(setParameters.GetItems()) > 0 {
+		reportCase(fpath, ALTER_TABLE_SET_ATTRUBUTE_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/1124",
+			"Remove it from the exported schema", "TABLE", fullyQualifiedName, sqlStmtInfo.stmt)
+	}
+
+	/*
+		e.g. alter table test add constraint uk unique(id) with (fillfactor='70');
+		alter_table_cmd:{subtype:AT_AddConstraint def:{constraint:{contype:CONSTR_UNIQUE conname:"asd" location:292
+		keys:{string:{sval:"id"}} options:{def_elem:{defname:"fillfactor" arg:{string:{sval:"70"}}...
+		Similarly here we are trying to get the constraint if any and then get the options field which is WITH options
+		in this case only so checking that for this case.
+	*/
+
+	if len(alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetDef().GetConstraint().GetOptions()) > 0 {
+		reportCase(fpath, STORAGE_PARAMETERS_DDL_STMT_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/23467",
+			"Remove the storage parameters from the DDL", "TABLE", fullyQualifiedName, sqlStmtInfo.stmt)
+	}
+
+	/*
+		e.g. ALTER TABLE example DISABLE example_rule;
+		cmds:{alter_table_cmd:{subtype:AT_DisableRule name:"example_rule" behavior:DROP_RESTRICT}} objtype:OBJECT_TABLE}}
+		checking the subType is sufficient in this case
+	*/
+	if alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetSubtype() == pg_query.AlterTableType_AT_DisableRule {
```

Comment:
is `ENABLE RULE` possible with pg_dump?

Reply:
No, `ENABLE RULE` is embedded in the `CREATE RULE DDL` itself as if the user has done disable -> enable on source then also only CREATE RULE will come.